service/worker/scheduler/workflow.go

```diff
@@ -1080,7 +1081,8 @@ func (s *scheduler) updateCustomSearchAttributes(searchAttributes *commonpb.Sear
 
 	//nolint:staticcheck // SA1019 Use untyped version for backwards compatibility.
 	currentSearchAttributes := workflow.GetInfo(s.ctx).SearchAttributes
-	for key, currentValuePayload := range currentSearchAttributes.GetIndexedFields() {
+	for _, key := range workflow.DeterministicKeys(currentSearchAttributes.GetIndexedFields()) {
```

Comment:
Same thing, the results are going in a map anyway so there's no reason to use DeterministicKeys

Reply:
The reason to use DeterministicKeys here (and in the other place) is to make the static checker happy. I agree that whether or not the map iteration is done with `range` or `DeterministicKeys` will not affect the overall workflow determinism. That is why I feel this change does not require a patch.

IMO, the reason to use DeterministicKeys is because it's "workflow best practice" to avoid _all_ range over maps, so that workflow authors don't have to think "does the ordering of this range affect my History Events" and then decide. It's better to just always require using DeterministicKeys and then never accidentally use `range` in a place that actually would affect determinism.

Reply:
workflowcheck (and other linters) are aids for finding bugs, not authorities. If they suggest something semantically equivalent but that makes the code harder to read (and also slower), we don't have to take the suggestion.

I get the point that it reduces mental overhead for someone reading the code, but we could also just disable workflowcheck here with a comment saying "iteration is safe because the results are going into a map".